store.rb

```diff
@@ -118,4 +153,26 @@ def get_node(node)
       raise e
     end
   end
+
+  # immediately update cache if node joins/leaves
+  def watch()
+    return if @zk.nil?
+    @root_watcher ||= @zk.register("/", :only => :child) do |event|
```

Comment:
I don't think `watch` is actually called again anywhere.  From what I half remember from zookeeper, watches have to be reregistered when they fire.